### ISSUE TITLE
Retrain model is not effective ,mAP aways is zero.

### ISSUE THREAD

**Issue by dao-kun** [tensorflow/models#7893]
**get tf.record：**

python object_detection/dataset_tools/create_pascal_tf_record_official.py \
    --label_map_path=object_detection/data/pascal_label_map.pbtxt \
    --data_dir=/home/cuda90/data/VOCdevkit/ --year=VOC2007 --set=train\
    --output_path=pascal_train.record

python object_detection/dataset_tools/create_pascal_tf_record.py \
    --label_map_path=object_detection/data/pascal_label_map.pbtxt \
    --data_dir=/home/cuda90/data/VOCdevkit/  --year=VOC2007 --set=val \
    --output_path=pascal_val.record

**OS Platform and Distribution：**

​	16.04.1-Ubuntu SMP Tue Nov 12 14:01:10 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

TensorFlow version：	 1.12

**GPU and and memory**：

​	 nvidia driver：NVIDIA-SMI 440.36，cuda：cuda9.0

​	nvidia card： TITAN Xp   *  4， memory： 12196MiB

**pipeline config**  ：

	model {
	ssd {
	    num_classes: 19    #changed
	    image_resizer {
	      fixed_shape_resizer {
	        height: 640
	        width: 640
	      }
	    }
	    feature_extractor {
	      type: "ssd_mobilenet_v1_fpn"
	      depth_multiplier: 1.0
	      min_depth: 16
	      conv_hyperparams {
	        regularizer {
	          l2_regularizer {
	            weight: 3.99999989895e-05
	          }
	        }
	        initializer {
	          random_normal_initializer {
	            mean: 0.0
	            stddev: 0.00999999977648
	          }
	        }
	        activation: RELU_6
	        batch_norm {
	          decay: 0.996999979019
	          scale: true
	          epsilon: 0.0010000000475
	        }
	      }
	      override_base_feature_extractor_hyperparams: true
	    }
	    box_coder {
	      faster_rcnn_box_coder {
	        y_scale: 10.0
	        x_scale: 10.0
	        height_scale: 5.0
	        width_scale: 5.0
	      }
	    }
	    matcher {
	      argmax_matcher {
	        matched_threshold: 0.5
	        unmatched_threshold: 0.5
	        ignore_thresholds: false
	        negatives_lower_than_unmatched: true
	        force_match_for_each_row: true
	        use_matmul_gather: true
	      }
	    }
	    similarity_calculator {
	      iou_similarity {
	      }
	    }
	    box_predictor {
	      weight_shared_convolutional_box_predictor {
	        conv_hyperparams {
	          regularizer {
	            l2_regularizer {
	              weight: 3.99999989895e-05
	            }
	          }
	          initializer {
	            random_normal_initializer {
	              mean: 0.0
	              stddev: 0.00999999977648
	            }
	          }
	          activation: RELU_6
	          batch_norm {
	            decay: 0.996999979019
	            scale: true
	            epsilon: 0.0010000000475
	          }
	        }
	        depth: 256
	        num_layers_before_predictor: 4
	        kernel_size: 3
	        class_prediction_bias_init: -4.59999990463
	      }
	    }
	    anchor_generator {
	      multiscale_anchor_generator {
	        min_level: 3
	        max_level: 7
	        anchor_scale: 4.0
	        aspect_ratios: 1.0
	        aspect_ratios: 2.0
	        aspect_ratios: 0.5
	        scales_per_octave: 2
	      }
	    }
	    post_processing {
	      batch_non_max_suppression {
	        score_threshold: 0.300000011921
	        iou_threshold: 0.600000023842
	        max_detections_per_class: 100
	        max_total_detections: 100
	      }
	      score_converter: SIGMOID
	    }
	    normalize_loss_by_num_matches: true
	    loss {
	      localization_loss {
	        weighted_smooth_l1 {
	        }
	      }
	      classification_loss {
	        weighted_sigmoid_focal {
	          gamma: 2.0
	          alpha: 0.25
	        }
	      }
	      classification_weight: 1.0
	      localization_weight: 1.0
	    }
	    encode_background_as_zeros: true
	    normalize_loc_loss_by_codesize: true
	    inplace_batchnorm_update: true
	    freeze_batchnorm: false
	  }
	}
	train_config {
	  batch_size:128
	  data_augmentation_options {
	    random_horizontal_flip {
	    }
	  }
	  data_augmentation_options {
	    random_crop_image {
	      min_object_covered: 0.0
	      min_aspect_ratio: 0.75
	      max_aspect_ratio: 3.0
	      min_area: 0.75
	      max_area: 1.0
	      overlap_thresh: 0.0
	    }
	  }
	  sync_replicas: true
	  optimizer {
	    momentum_optimizer {
	      learning_rate {
	        cosine_decay_learning_rate {
	          learning_rate_base: 0.0799999982119
	          total_steps: 12500
	          warmup_learning_rate:0.0266660004854
	          warmup_steps: 1000
	        }
	      }
	      momentum_optimizer_value: 0.899999976158
	    }
	    use_moving_average: false
	  }
	  fine_tune_checkpoint:  "/home/cuda90/daokun/models-master/research/object_detection/pretrained_model/ssd_mobilenet_v1_fpn_shared_box_predictor_640x640_coco14_sync_2018_07_03/model.ckpt"  # changed
	  num_steps: 12500
	  startup_delay_steps: 0.0
	  replicas_to_aggregate: 8
	  max_number_of_boxes: 100
	  unpad_groundtruth_tensors: false
	}
	train_input_reader {
	  label_map_path: "./data/pascal_label_map_sign.pbtxt"    #changed
	  tf_record_input_reader {
	    input_path: "./data/pascal_train_sign.record"	 #changed
	  }
	}
	eval_config {
	  num_examples: 8000
	  metrics_set: "coco_detection_metrics"
	  use_moving_averages: false
	}
	eval_input_reader {
	  label_map_path: "./data/pascal_label_map_sign.pbtxt"	 #changed
	  shuffle: false
	  num_readers: 1
	  tf_record_input_reader {
	    input_path: "./data/pascal_val_sign.record"  #changed
	  }
	}

  **Exact command to  train：**

​		

#From tensorflow/models/research/

./bin/protoc object_detection/protos/*.proto --python_out=.

#From tensorflow/models/research/

export PYTHONPATH=$PYTHONPATH:`pwd`:`pwd`/slim

PIPELINE_CONFIG_PATH=/home/cuda90/daokun/models-master/research/models/mobilenetv1/pipeline.config
MODEL_DIR=./models/mobilenetv1_pipeline_new
NUM_TRAIN_STEPS=50000
SAMPLE_1_OF_N_EVAL_EXAMPLES=1

python  object_detection/model_main.py \
    --pipeline_config_path=${PIPELINE_CONFIG_PATH} \
    --model_dir=${MODEL_DIR} \
    --num_train_steps=${NUM_TRAIN_STEPS} \
    --sample_1_of_n_eval_examples=$SAMPLE_1_OF_N_EVAL_EXAMPLES \
    --alsologtostderr

**result：** 
creating index...
index created!
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=4.35s).
Accumulating evaluation results...
DONE (t=1.73s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
### Describe the problem

I'm trying to fiturn the model(ssd_moblenet_v2_coco or ssd_resnet_50_fpn_coco) and want to  depoly it on Android device, the fineturn checkpoint and pipeline config that is given in the downloaded model from the model zoo, but no matter how many epochs I train, the mAP is always 0. Has anyone successfully trained the model?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
TensorFlow installed from
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by dao-kun** [comment]
I I have figured out this problem,thanks!

**Comment by hoang-ho** [comment]
How did you fix it?

**Comment by dao-kun** [comment]
I just replaced it with another configfile :object_detection/samples/ssd_mobilenet_v1_coco.config, then it works,may be this config is wrong.

**Comment by yzmean** [comment]
Have the same issue ：(

**Comment by yzmean** [comment]
@srjoglekar246 Sorry to bother you again, could you give us any suggestion :)

**Comment by hoang-ho** [comment]
Solved by:

```
load_all_detection_checkpoint_vars: true
```

**Comment by dao-kun** [comment]
> Solved by adding the following line in config file:
> 
> ```
> load_all_detection_checkpoint_vars: true
> ```

Thanks, I'll try again~
